test/activity_heartbeating/my_workflow_test.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require 'test'
+require 'activity_heartbeating/my_workflow'
+require 'securerandom'
+require 'temporalio/testing'
+require 'temporalio/worker'
+
+module ActivityHeartbeating
+  class ActivityHeartbeatingTest < Test
+    def test_workflow
+      # Run test server until completion of the block
+      Temporalio::Testing::WorkflowEnvironment.start_local do |env|
+        # Run worker until completion of the block
+        worker = Temporalio::Worker.new(
+          client: env.client,
+          task_queue: "tq-#{SecureRandom.uuid}",
+          activities: [MyActivities::FakeProgress],
+          workflows: [MyWorkflow]
+        )
+        worker.run do
+          # Start workflow
+          wf = env.client.start_workflow(MyWorkflow, id: "wf-#{SecureRandom.uuid}", task_queue: worker.task_queue)
+
+          # Cancel workflow
+          wf.cancel
```

Comment:
A common approach we do internally is wait until the activity has started (by continually describing until the pending activity is present or at least continually polling history continually waiting for activity scheduled event). This behavior may actually never have the workflow reach the worker. But this is non-blocking.

Reply:
Done, reads very nicely in Ruby :)